contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/GobblinDistcpJob.java

```diff
@@ -0,0 +1,195 @@
+        Path path = new Path(target.replaceAll(".*://.*?(?=/)", ""));
+
+        if (fs.exists(path)) {
+            fs.delete(path, true);
+            if (fs.exists(path)) {
+                warn("Could not delete temp directory " + path + " in CDN!");
+            } else {
+                info("Deleted " + path);
+            }
+        }
+    }
+
+    private void deleteDirOnExit(FileSystem fs, String target) throws Exception {
+        Path path = new Path(target.replaceAll(".*://.*?(?=/)", ""));
+
+        fs.deleteOnExit(path);  // Delete the directory even if an exception occurs
+        info(path + " is scheduled to be deleted on exit.");
+    }
+
+    private void addPermissionsToParents(FileSystem fs, String target) throws Exception {
+        String pathFromRoot = target.replaceAll(".*://.*?/", "");
+        String parent = "";
+
+        for (String seg: pathFromRoot.split("/")) {
+            parent = parent + "/" + seg;
+            Path parentPath = new Path(parent);
+            FsPermission perm = fs.getFileStatus(parentPath).getPermission();
+            FsAction u = perm.getUserAction();
+            FsAction g = perm.getGroupAction();
+            FsAction o = perm.getOtherAction();
+            boolean changed = false;
+
+            if (props.getBoolean(VoldemortBuildAndPushJob.PUSH_CDN_READBYGROUP, false)) {
+                if (!g.implies(FsAction.READ_EXECUTE)) {
+                    g = g.or(FsAction.READ_EXECUTE);
+                    changed = true;
+                }
+            }
+
+            if (props.getBoolean(VoldemortBuildAndPushJob.PUSH_CDN_READBYOTHER, false)) {
+                if (!o.implies(FsAction.READ_EXECUTE)) {
+                    o = o.or(FsAction.READ_EXECUTE);
+                    changed = true;
+                }
+            }
+
+            if (changed) {
+                FsPermission desiredPerm = new FsPermission(u, g, o);
+                fs.setPermission(parentPath, desiredPerm);
+                assert fs.getFileStatus(parentPath).getPermission().equals(desiredPerm);
+                info("for path " + parent + ", permissions changed:");
+                info("from: " + perm);
+                info("  to: " + desiredPerm);
+            }
+        }
+    }
+
+    private String pickCDN() {
+        List<String> pushClusters = props.getList(VoldemortBuildAndPushJob.PUSH_CLUSTER);
+
+        if (!props.containsKey(VoldemortBuildAndPushJob.PUSH_CDN_CLUSTER)) {
+            return "";
+        }
+        List<String> cdnClusters = props.getList(VoldemortBuildAndPushJob.PUSH_CDN_CLUSTER);
+
+        if (pushClusters.size() != cdnClusters.size()) {
+            warn("Cluster sizes are different! Will bypass CDN for push cluster " + destination);
+            return "";
+        }
+
+        int index = pushClusters.indexOf(destination);
+        assert index != -1;
+        String cdnCluster = cdnClusters.get(index);
+        if (cdnCluster.equals("null")) {
```

Comment:
In that case we will need a new parser. Since the setup is a one-off task, it doesn't make a lot of difference in my opinion. I would like to keep the current implementation.